storage/src/main/java/org/commonjava/storage/pathmapped/core/PathMappedFileManager.java

```diff
@@ -116,6 +116,41 @@ public InputStream openInputStream( String fileSystem, String path ) throws IOEx
         return stream;
     }
 
+    /**
+     * Get storage file if not expired. Delete the file and return null if expired.
+     * If 'resetTimeoutForAccessing' configured, reset timeout when the expiration is close to due date. This is
+     * to prevent commonly used files being expired too soon.
+     */
+    private String getStorageFileInternal( String fileSystem, String path )
+    {
+        final PathMap pathMap = getPathMap( fileSystem, path );
+        if ( pathMap != null )
+        {
+            final Date expiration = pathMap.getExpiration();
+            if ( expiration != null )
+            {
+                final long current = System.currentTimeMillis();
+                if ( expiration.getTime() < current ) // timeout
+                {
+                    logger.info("File expired, filesystem: {}, path: {}, expiration: {}", fileSystem, path, expiration);
+                    delete(fileSystem, path);
+                    return null;
```

Comment:
It will be deleted and returns null, but why it still reports missing physical file before. Just confirm the logic. ; -)

Reply:
Here null means the entry expired in DB. The missing physical file is different, which in the DB it has a valid entry, but 'physicalStore.getInputStream( storageFile );' returns null.

Reply:
if we hit 'missing physical file', it is a real bug. like what we experienced before, the wrong dedup causing physical file being deleted.

Reply:
Yeah, I see.